Kudu.Core/Infrastructure/ProcessExtensions.cs

```diff
@@ -244,7 +244,14 @@ public static string GetCommandLine(this Process process)
 
         public static bool GetIsScmSite(Dictionary<string, string> environment)
         {
-            return environment[WellKnownEnvironmentVariables.ApplicationPoolId].StartsWith("~1", StringComparison.OrdinalIgnoreCase);
+            string appPool = null;
+            if (environment.TryGetValue(WellKnownEnvironmentVariables.ApplicationPoolId, out appPool) &&
+                !string.IsNullOrEmpty(appPool))
```

Comment:
if environments contains a key with null value its a bug and we would actually want to break on that.